components/dialog/test/dialog.vdiff.js

```diff
@@ -100,6 +125,23 @@ describe('dialog', () => {
 				await expect(document).to.be.golden();
 			});
 
+			it('delayed-content', async() => {
+				const el = await fixture(
+					`<d2l-dialog title-text="Delayed Dialog"><${delayedTag}></${delayedTag}></d2l-dialog>`,
+					{
+						awaitLoadingComplete: false
+					}
+				);
+				setTimeout(async() => {
+					await waitUntil(() => el.querySelector(delayedTag) !== null);
+					setTimeout(() => el.querySelector(delayedTag).finishLoading(), 100);
+				});
+				el.opened = true;
+				await oneEvent(el, 'd2l-dialog-open');
```

Comment:
The way this is going to work right now:
1. Timeout fires but nothing happens yet
2. Dialog opens, starts its opening logic
3. Once the opening logic is done and the initial resize happens, the event will fire
4. Screenshot is taken
5. Sometime later, the inner timeout fires which causes the delayed element to finish loading -- but it's too late

Once this is fixed, the sequence will be:
1. Timeout fires but nothing happens yet
2. Dialog opens, starts its opening logic
3. It's going to wait for everything inside it to be "loading complete"
4. The inner timeout fires which causes the delayed element to finish loading
5. Resize happens, the event will fire
6. Screenshot is taken